### PR TITLE
Se 342 set up diamond test through echidna

### PR DESCRIPTION
**Pull request by ShaneDuncan602**
This is just a simple example of creating the RuleStorageDiamond and property testing the version facet.

### CODE REVIEW

**Review by TJ-Everett — APPROVED** [review]
Somehow the two files changed for the PR doesn't seem to reflect the amount of work/frustration I know when in to get Echidna working.

File: test/echidna/helpers/TestCommonEchidna.sol [hunk 1]
```diff
@@ -1,31 +1,253 @@
...
+        // deploy the facet
+        assembly {
+            _facetAddress := create(0, add(_bytecode, 0x20), mload(_bytecode))
+        }
+
+        // Get the facet selectors.
+        inputs[2] = _facetName;
+        bytes memory res = vm.ffi(inputs);
+        _selectors = abi.decode(res, (bytes4[]));
+        return (_facetAddress, _selectors);
```
Comment (oscarsernarosero) [inline test/echidna/helpers/TestCommonEchidna.sol:119]: So the fix was just to add the `end=""` ?

File: test/echidna/property/RuleStorageDiamond.e.sol [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    constructor() {
+        ruleStorageDiamond = _createRuleStorageDiamond();
+        VersionFacet(address(ruleStorageDiamond)).updateVersion("1,0,0");
+    }
+
+    /* ------------------------------ INVARIANTS -------------------------------- */
+    /// Test the version
+    function echidna_version_notblank() public returns (bool) {
+        return (bytes(VersionFacet(address(ruleStorageDiamond)).version()).length != 0);
```
Comment (oscarsernarosero) [inline test/echidna/property/RuleStorageDiamond.e.sol]: Should we test more functionality in the diamond? like upgrades, adding a rule and stuff like that?

**Review by oscarsernarosero — APPROVED** [review]
Looks really good Shane. Great job. Im just curious if you think the test on the diamond is enough as it is.

**Comment by ShaneDuncan602** [comment]
> Looks really good Shane. Great job. Im just curious if you think the test on the diamond is enough as it is.

This was just to give an example of how we can build a diamond and test. We still have to get together and decide the best way to do it.

File: test/echidna/property/RuleStorageDiamond.e.sol (hunk 2 above)
Comment (ShaneDuncan602) [reply test/echidna/property/RuleStorageDiamond.e.sol]: This is invariant testing. It only tests the properties. We could add assertion testing in the future.

**Pull request merged**